src/functions/updatePackageVersions.js

```diff
@@ -20,20 +37,33 @@ export default async function updatePackageVersions(
   let cwd = opts.cwd || process.cwd();
   let project = await Project.init(cwd);
   let workspaces = await project.getWorkspaces();
+  let { graph } = await project.getDependencyGraph(workspaces);
   let updatedPackages = new Set();
 
+  const internalDeps = Object.keys(versions).filter(dep => graph.has(dep));
+  const externalDeps = Object.keys(versions).filter(dep => !graph.has(dep));
+
+  if (externalDeps.length !== 0) {
+    logger.warn(
+      messages.externalDepsPassedToUpdatePackageVersions(externalDeps)
+    );
+  }
+
   for (let workspace of workspaces) {
     let pkg = workspace.pkg;
     let promises = [];
 
-    for (let depName of Object.keys(versions)) {
-      let depTypes = pkg.getDependencyTypes(depName);
+    for (let depName of internalDeps) {
+      const depRange = String(pkg.getDependencyVersionRange(depName));
+      const depTypes = pkg.getDependencyTypes(depName);
+      const rangeType = versionRangeToRangeType(depRange);
+
       if (depTypes.length === 0) continue;
       for (let depType of depTypes) {
         await pkg.setDependencyVersionRange(
           depName,
           depType,
-          versions[depName]
+          rangeType + versions[depName]
```

Comment:
Not a huge fan of doing actions inside a function call's arguments.

Reply:
agreen